openmls/src/group/core_group/new_from_external_init.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{
+    ciphersuite::signable::Verifiable,
+    credentials::CredentialBundle,
+    framing::plaintext::MlsPlaintext,
+    group::errors::ExternalInitError,
+    key_packages::KeyPackageBundle,
+    messages::{
+        proposals::{ExternalInitProposal, Proposal},
+        public_group_state::{PublicGroupState, VerifiablePublicGroupState},
+    },
+};
+
+use super::{
+    create_commit_params::{CommitType, CreateCommitParams},
+    proposals::{ProposalStore, StagedProposal},
+    CoreGroup,
+};
+use crate::group::core_group::*;
+
+pub type ExternalInitResult = Result<
+    (
+        CoreGroup,
+        MlsPlaintext,
+        Option<Welcome>,
+        Option<KeyPackageBundle>,
+    ),
+    ExternalInitError,
+>;
+
+impl CoreGroup {
+    /// Join a group without the help of an internal member. This function
+    /// requires a `PublicGroupState`, as well as the corresponding public tree
+    /// `nodes`. After the group state is initialized, this function creates an
+    /// `ExternalInit` proposal and commits it along with the given proposals by
+    /// reference and by value.
+    ///
+    /// Returns the new `CoreGroup` object, as well as the `MlsPlaintext`
+    /// containing the commit.
+    pub fn new_from_external_init(
+        backend: &impl OpenMlsCryptoProvider,
+        framing_parameters: FramingParameters,
+        nodes_option: Option<Vec<Option<Node>>>,
```

Comment:
I think `Option` is not a bad choice here, since it make it explicit that providing a tree is optional, i.e. one can rely exclusively on the RatchetTreeExtension. You're right regarding the name, though.